src/machine/machined-varlink.c

```diff
@@ -388,14 +388,66 @@ static int vl_method_get_memberships(sd_varlink *link, sd_json_variant *paramete
         return sd_varlink_error(link, "io.systemd.UserDatabase.NoRecordFound", NULL);
 }
 
-static int list_machine_one(sd_varlink *link, Machine *m, bool more) {
-        int r;
+typedef enum ListMachineFlags {
+        LIST_MACHINE_NEED_MODE          = 1 << 0,
+        LIST_MACHINE_ACQUIRE_ADDRESSES  = 1 << 1,
+        LIST_MACHINE_ACQUIRE_OS_RELEASE = 1 << 2,
+        LIST_MACHINE_ACQUIRE_UID_SHIFT  = 1 << 3,
+} ListMachineFlags;
+
+static int list_machine_one_and_maybe_read_metadata(sd_varlink *link, Machine *m, ListMachineFlags flags) {
+        _cleanup_free_ struct local_address *addresses = NULL;
+        _cleanup_strv_free_ char **os_release = NULL;
+        uid_t shift = UID_INVALID;
+        int n, r;
 
         assert(link);
         assert(m);
 
         _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
 
+        /* TODO(ikruglov): maybe swallow errors and don't output fields if NoPrivateNetwork/NoOSReleaseInformation or NotSupported?
+         * I think swallowing errors in general listing (varlinkctl --more call ... io.systemd.Machine.List '{}') make sense.
+         * However, when a user request information about specific machine (varlinkctl --more call ... io.systemd.Machine.List '{"name": "foo"}')
+         * giving a error is better */
+        if (FLAGS_SET(flags, LIST_MACHINE_ACQUIRE_ADDRESSES)) {
+                n = machine_get_addresses(m, &addresses);
+                if (n == -ENONET)
+                        return sd_varlink_error(link, "io.systemd.Machine.NoPrivateNetworking", NULL);
+                if (ERRNO_IS_NEG_NOT_SUPPORTED(n))
+                        return sd_varlink_errorbo(link,
+                                        "io.systemd.Machine.NotSupported",
+                                        SD_JSON_BUILD_PAIR_STRING("reason", "Requesting IP address data is only supported on container machines"));
```

Comment:
nah, please don't do that. no english text in error codes please.

Reply:
maybe generate an error io.systemd.Machine.DataNotAvailableForMachineClass?

Reply:
what do you want to see? io.systemd.Machine.NotSupportedForNotContainer?